platform/preferred_languages.cpp

```diff
@@ -169,6 +172,41 @@ std::string GetCurrentNorm()
   return Normalize(GetCurrentOrig());
 }
 
+std::string GetUtf8MultilangSupported()
+{
+  std::string languageCode;
+
+  auto const & systemLanguages = GetSystemPreferred();
+  for (const auto & systemLanguage: systemLanguages)
+  {
+    const std::string normalizedLang = Normalize(systemLanguage);
+    if (StringUtf8Multilang::GetLangIndex(normalizedLang) != StringUtf8Multilang::kUnsupportedLanguageCode)
+    {
+      languageCode = normalizedLang;
+      break;
+    }
+  }
+
+  if (languageCode.empty())
+    throw std::runtime_error("Unsupported language.");
+
+  return languageCode;
+}
+
+std::string GetCurrentMapLanguageCode()
+{
+  std::string languageCode;
+  if (!settings::Get(settings::kMapLanguageCode, languageCode) || languageCode.empty())
+    languageCode = GetUtf8MultilangSupported();
```

Comment:
1. The name GetUtf8MultilangSupported is not great.
2. The code of GetUtf8MultilangSupported function can be directly placed here.

Reply:
removed & done